File: Calculadora.cs
using System;

class CalculadoraCientifica
{
    static void Main()
    {
        int opcao;
        
        do
        {
            MostrarMenuPrincipal();
            opcao = int.Parse(Console.ReadLine());
            
            switch (opcao)
            {
                case 1:
                    MenuBasicas();
                    break;
                case 2:
                    MenuPotenciaRaiz();
                    break;
                case 3:
                    MenuTrigonometria();
                    break;
                case 4:
                    MenuLogaritmos();
                    break;
                case 5:
                    MenuComplexos();
                    break;
                case 0:
                    Console.WriteLine("\nSaindo... Obrigado por usar a calculadora!");
                    break;
                default:
                    Console.WriteLine("Opção inválida!");
                    Pausa();
                    break;
            }
            
        } while (opcao != 0);
    }

    static void MostrarMenuPrincipal()
    {
        Console.Clear();
        Console.WriteLine("================================================");
        Console.WriteLine("         CALCULADORA CIENTÍFICA (C#)");
        Console.WriteLine("================================================");
        Console.WriteLine("1. Operações Básicas");
        Console.WriteLine("2. Potência e Raízes");
        Console.WriteLine("3. Trigonometria");
        Console.WriteLine("4. Logaritmos");
        Console.WriteLine("5. Números Complexos");
        Console.WriteLine("0. Sair");
        Console.WriteLine("================================================");
        Console.Write("Escolha uma opção: ");
    }

    static void MenuBasicas()
    {
        int op;
        do
        {
            Console.Clear();
            Console.WriteLine("--- OPERAÇÕES BÁSICAS ---");
            Console.WriteLine("1. Adição");
            Console.WriteLine("2. Subtração");
            Console.WriteLine("3. Multiplicação");
            Console.WriteLine("4. Divisão");
            Console.WriteLine("0. Voltar");
            Console.Write("Escolha: ");
            op = int.Parse(Console.ReadLine());

            if (op >= 1 && op <= 4)
            {
                Console.Write("Digite o primeiro número: ");
                double a = double.Parse(Console.ReadLine());
                Console.Write("Digite o segundo número: ");
                double b = double.Parse(Console.ReadLine());

                switch (op)
                {
                    case 1:
                        Console.WriteLine($"Resultado: {a} + {b} = {a + b}");
                        break;
                    case 2:
                        Console.WriteLine($"Resultado: {a} - {b} = {a - b}");
                        break;
                    case 3:
                        Console.WriteLine($"Resultado: {a} × {b} = {a * b}");
                        break;
                    case 4:
                        if (b == 0)
                            Console.WriteLine("Erro: Divisão por zero não permitida!");
                        else
                            Console.WriteLine($"Resultado: {a} ÷ {b} = {a / b}");
                        break;
                }
            }
            else if (op != 0)
            {
                Console.WriteLine("Opção inválida!");
            }

            if (op != 0) Pausa();

        } while (op != 0);
    }

    static void MenuPotenciaRaiz()
    {
        int op;
        do
        {
            Console.Clear();
            Console.WriteLine("--- POTÊNCIA E RAÍZES ---");
            Console.WriteLine("1. Potência (x^y)");
            Console.WriteLine("2. Raiz Quadrada");
            Console.WriteLine("3. Raiz Cúbica");
            Console.WriteLine("0. Voltar");
            Console.Write("Escolha: ");
            op = int.Parse(Console.ReadLine());

            switch (op)
            {
                case 1:
                    Console.Write("Digite a base: ");
                    double a = double.Parse(Console.ReadLine());
                    Console.Write("Digite o expoente: ");
                    double b = double.Parse(Console.ReadLine());
                    Console.WriteLine($"Resultado: {a}^{b} = {Math.Pow(a, b)}");
                    break;
                case 2:
                    Console.Write("Digite o número: ");
                    a = double.Parse(Console.ReadLine());
                    if (a < 0)
                        Console.WriteLine("Erro: Raiz quadrada de número negativo não é real!");
                    else
                        Console.WriteLine($"Resultado: √{a} = {Math.Sqrt(a)}");
                    break;
                case 3:
                    Console.Write("Digite o número: ");
                    a = double.Parse(Console.ReadLine());
                    if (a < 0)
                        Console.WriteLine($"Resultado: ∛{a} = -{Math.Pow(-a, 1.0/3.0)}");
                    else
                        Console.WriteLine($"Resultado: ∛{a} = {Math.Pow(a, 1.0/3.0)}");
                    break;
                case 0:
                    break;
                default:
                    Console.WriteLine("Opção inválida!");
                    break;
            }

            if (op != 0) Pausa();

        } while (op != 0);
    }

    static void MenuTrigonometria()
    {
        int op;
        do
        {
            Console.Clear();
            Console.WriteLine("--- TRIGONOMETRIA (ângulos em graus) ---");
            Console.WriteLine("1. Seno");
            Console.WriteLine("2. Cosseno");
            Console.WriteLine("3. Tangente");
            Console.WriteLine("4. Arco Seno (inverso)");
            Console.WriteLine("5. Arco Cosseno (inverso)");
            Console.WriteLine("0. Voltar");
            Console.Write("Escolha: ");
            op = int.Parse(Console.ReadLine());

            switch (op)
            {
                case 1:
                case 2:
                case 3:
                    Console.Write("Digite o ângulo em graus: ");
                    double ang = double.Parse(Console.ReadLine());
                    double rad = ang * Math.PI / 180;
                    
                    if (op == 1)
                        Console.WriteLine($"sen({ang}°) = {Math.Sin(rad):F4}");
                    else if (op == 2)
                        Console.WriteLine($"cos({ang}°) = {Math.Cos(rad):F4}");
                    else
                        Console.WriteLine($"tan({ang}°) = {Math.Tan(rad):F4}");
                    break;
                
                case 4:
                    Console.Write("Digite o valor (entre -1 e 1): ");
                    double val = double.Parse(Console.ReadLine());
                    if (val >= -1 && val <= 1)
                        Console.WriteLine($"arcsen({val}) = {Math.Asin(val) * 180 / Math.PI:F2}°");
                    else
                        Console.WriteLine("Erro: Valor fora do domínio (-1 a 1)");
                    break;
                
                case 5:
                    Console.Write("Digite o valor (entre -1 e 1): ");
                    val = double.Parse(Console.ReadLine());
                    if (val >= -1 && val <= 1)
                        Console.WriteLine($"arccos({val}) = {Math.Acos(val) * 180 / Math.PI:F2}°");
                    else
                        Console.WriteLine("Erro: Valor fora do domínio (-1 a 1)");
                    break;
                
                case 0:
                    break;
                
                default:
                    Console.WriteLine("Opção inválida!");
                    break;
            }

            if (op != 0) Pausa();

        } while (op != 0);
    }

    static void MenuLogaritmos()
    {
        int op;
        do
        {
            Console.Clear();
            Console.WriteLine("--- LOGARITMOS ---");
            Console.WriteLine("1. Logaritmo Natural (ln)");
            Console.WriteLine("2. Logaritmo Base 10 (log10)");
            Console.WriteLine("3. Logaritmo em qualquer base");
            Console.WriteLine("0. Voltar");
            Console.Write("Escolha: ");
            op = int.Parse(Console.ReadLine());

            switch (op)
            {
                case 1:
                    Console.Write("Digite o número: ");
                    double a = double.Parse(Console.ReadLine());
                    if (a <= 0)
                        Console.WriteLine("Erro: Logaritmo de número não positivo!");
                    else
                        Console.WriteLine($"ln({a}) = {Math.Log(a):F4}");
                    break;
                
                case 2:
                    Console.Write("Digite o número: ");
                    a = double.Parse(Console.ReadLine());
                    if (a <= 0)
                        Console.WriteLine("Erro: Logaritmo de número não positivo!");
                    else
                        Console.WriteLine($"log10({a}) = {Math.Log10(a):F4}");
                    break;
                
                case 3:
                    Console.Write("Digite o número: ");
                    a = double.Parse(Console.ReadLine());
                    Console.Write("Digite a base: ");
                    double baseLog = double.Parse(Console.ReadLine());
                    
                    if (a <= 0 || baseLog <= 0 || baseLog == 1)
                        Console.WriteLine("Erro: Parâmetros inválidos!");
                    else
                        Console.WriteLine($"log{baseLog}({a}) = {Math.Log(a, baseLog):F4}");
                    break;
                
                case 0:
                    break;
                
                default:
                    Console.WriteLine("Opção inválida!");
                    break;
            }

            if (op != 0) Pausa();

        } while (op != 0);
    }

    static void MenuComplexos()
    {
        int op;
        do
        {
            Console.Clear();
            Console.WriteLine("--- NÚMEROS COMPLEXOS ---");
            Console.WriteLine("1. Adição (a + bi) + (c + di)");
            Console.WriteLine("2. Multiplicação");
            Console.WriteLine("3. Módulo");
            Console.WriteLine("4. Forma Polar");
            Console.WriteLine("0. Voltar");
            Console.Write("Escolha: ");
            op = int.Parse(Console.ReadLine());

            switch (op)
            {
                case 1:
                    Console.Write("Parte real do primeiro: ");
                    double a = double.Parse(Console.ReadLine());
                    Console.Write("Parte imaginária do primeiro: ");
                    double b = double.Parse(Console.ReadLine());
                    Console.Write("Parte real do segundo: ");
                    double c = double.Parse(Console.ReadLine());
                    Console.Write("Parte imaginária do segundo: ");
                    double d = double.Parse(Console.ReadLine());
                    
                    Console.WriteLine($"Resultado: ({a} + {b}i) + ({c} + {d}i) = {a + c} + {b + d}i");
                    break;
                
                case 2:
                    Console.Write("Parte real do primeiro: ");
                    a = double.Parse(Console.ReadLine());
                    Console.Write("Parte imaginária do primeiro: ");
                    b = double.Parse(Console.ReadLine());
                    Console.Write("Parte real do segundo: ");
                    c = double.Parse(Console.ReadLine());
                    Console.Write("Parte imaginária do segundo: ");
                    d = double.Parse(Console.ReadLine());
                    
                    double real = a * c - b * d;
                    double imag = a * d + b * c;
                    Console.WriteLine($"Resultado: ({a} + {b}i) × ({c} + {d}i) = {real} + {imag}i");
                    break;
                
                case 3:
                    Console.Write("Parte real: ");
                    a = double.Parse(Console.ReadLine());
                    Console.Write("Parte imaginária: ");
                    b = double.Parse(Console.ReadLine());
                    
                    double modulo = Math.Sqrt(a * a + b * b);
                    Console.WriteLine($"Módulo de {a} + {b}i = {modulo:F4}");
                    break;
                
                case 4:
                    Console.Write("Parte real: ");
                    a = double.Parse(Console.ReadLine());
                    Console.Write("Parte imaginária: ");
                    b = double.Parse(Console.ReadLine());
                    
                    modulo = Math.Sqrt(a * a + b * b);
                    double fase = Math.Atan2(b, a) * 180 / Math.PI;
                    Console.WriteLine($"Forma polar: {modulo:F4} ∠ {fase:F2}°");
                    break;
                
                case 0:
                    break;
                
                default:
                    Console.WriteLine("Opção inválida!");
                    break;
            }

            if (op != 0) Pausa();

        } while (op != 0);
    }

    static void Pausa()
    {
        Console.WriteLine("\nPressione qualquer tecla para continuar...");
        Console.ReadKey();
    }
}

File: calculadora.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function perguntar(pergunta) {
    return new Promise((resolve) => {
        rl.question(pergunta, resolve);
    });
}

function mostrarMenuPrincipal() {
    console.log("\n" + "=".repeat(50));
    console.log("         CALCULADORA CIENTÍFICA (JavaScript)");
    console.log("=".repeat(50));
    console.log("1. Operações Básicas");
    console.log("2. Potência e Raízes");
    console.log("3. Trigonometria");
    console.log("4. Logaritmos");
    console.log("5. Números Complexos");
    console.log("0. Sair");
    console.log("=".repeat(50));
}

async function menuBasicas() {
    let opcao;
    do {
        console.clear();
        console.log("--- OPERAÇÕES BÁSICAS ---");
        console.log("1. Adição");
        console.log("2. Subtração");
        console.log("3. Multiplicação");
        console.log("4. Divisão");
        console.log("0. Voltar");
        
        opcao = await perguntar("Escolha: ");
        
        if (opcao === '1') {
            let num1 = parseFloat(await perguntar("Digite o primeiro número: "));
            let num2 = parseFloat(await perguntar("Digite o segundo número: "));
            console.log(`Resultado: ${num1} + ${num2} = ${num1 + num2}`);
        } else if (opcao === '2') {
            let num1 = parseFloat(await perguntar("Digite o primeiro número: "));
            let num2 = parseFloat(await perguntar("Digite o segundo número: "));
            console.log(`Resultado: ${num1} - ${num2} = ${num1 - num2}`);
        } else if (opcao === '3') {
            let num1 = parseFloat(await perguntar("Digite o primeiro número: "));
            let num2 = parseFloat(await perguntar("Digite o segundo número: "));
            console.log(`Resultado: ${num1} × ${num2} = ${num1 * num2}`);
        } else if (opcao === '4') {
            let num1 = parseFloat(await perguntar("Digite o primeiro número: "));
            let num2 = parseFloat(await perguntar("Digite o segundo número: "));
            if (num2 === 0) {
                console.log("Erro: Divisão por zero não permitida!");
            } else {
                console.log(`Resultado: ${num1} ÷ ${num2} = ${num1 / num2}`);
            }
        }
        
        if (opcao !== '0') {
            await perguntar("\nPressione Enter para continuar...");
        }
    } while (opcao !== '0');
}

async function menuPotenciaRaiz() {
    let opcao;
    do {
        console.clear();
        console.log("--- POTÊNCIA E RAÍZES ---");
        console.log("1. Potência (x^y)");
        console.log("2. Raiz Quadrada");
        console.log("3. Raiz Cúbica");
        console.log("0. Voltar");
        
        opcao = await perguntar("Escolha: ");
        
        if (opcao === '1') {
            let base = parseFloat(await perguntar("Digite a base: "));
            let expoente = parseFloat(await perguntar("Digite o expoente: "));
            console.log(`Resultado: ${base}^${expoente} = ${Math.pow(base, expoente)}`);
        } else if (opcao === '2') {
            let numero = parseFloat(await perguntar("Digite o número: "));
            if (numero < 0) {
                console.log("Erro: Raiz quadrada de número negativo não é real!");
            } else {
                console.log(`Resultado: √${numero} = ${Math.sqrt(numero)}`);
            }
        } else if (opcao === '3') {
            let numero = parseFloat(await perguntar("Digite o número: "));
            if (numero < 0) {
                console.log(`Resultado: ∛${numero} = -${Math.pow(-numero, 1/3)}`);
            } else {
                console.log(`Resultado: ∛${numero} = ${Math.pow(numero, 1/3)}`);
            }
        }
        
        if (opcao !== '0') {
            await perguntar("\nPressione Enter para continuar...");
        }
    } while (opcao !== '0');
}

async function menuTrigonometria() {
    let opcao;
    do {
        console.clear();
        console.log("--- TRIGONOMETRIA ---");
        console.log("1. Seno");
        console.log("2. Cosseno");
        console.log("3. Tangente");
        console.log("4. Arco Seno");
        console.log("5. Arco Cosseno");
        console.log("6. Arco Tangente");
        console.log("0. Voltar");
        
        opcao = await perguntar("Escolha: ");
        
        if (opcao === '1') {
            let angulo = parseFloat(await perguntar("Digite o ângulo em graus: "));
            let radianos = angulo * Math.PI / 180;
            console.log(`sen(${angulo}°) = ${Math.sin(radianos).toFixed(4)}`);
        } else if (opcao === '2') {
            let angulo = parseFloat(await perguntar("Digite o ângulo em graus: "));
            let radianos = angulo * Math.PI / 180;
            console.log(`cos(${angulo}°) = ${Math.cos(radianos).toFixed(4)}`);
        } else if (opcao === '3') {
            let angulo = parseFloat(await perguntar("Digite o ângulo em graus: "));
            let radianos = angulo * Math.PI / 180;
            console.log(`tan(${angulo}°) = ${Math.tan(radianos).toFixed(4)}`);
        } else if (opcao === '4') {
            let valor = parseFloat(await perguntar("Digite o valor (entre -1 e 1): "));
            if (valor >= -1 && valor <= 1) {
                console.log(`arcsen(${valor}) = ${(Math.asin(valor) * 180 / Math.PI).toFixed(2)}°`);
            } else {
                console.log("Erro: Valor fora do domínio (-1 a 1)");
            }
        } else if (opcao === '5') {
            let valor = parseFloat(await perguntar("Digite o valor (entre -1 e 1): "));
            if (valor >= -1 && valor <= 1) {
                console.log(`arccos(${valor}) = ${(Math.acos(valor) * 180 / Math.PI).toFixed(2)}°`);
            } else {
                console.log("Erro: Valor fora do domínio (-1 a 1)");
            }
        } else if (opcao === '6') {
            let valor = parseFloat(await perguntar("Digite o valor: "));
            console.log(`arctan(${valor}) = ${(Math.atan(valor) * 180 / Math.PI).toFixed(2)}°`);
        }
        
        if (opcao !== '0') {
            await perguntar("\nPressione Enter para continuar...");
        }
    } while (opcao !== '0');
}

async function menuLogaritmos() {
    let opcao;
    do {
        console.clear();
        console.log("--- LOGARITMOS ---");
        console.log("1. Logaritmo Natural (ln)");
        console.log("2. Logaritmo Base 10 (log10)");
        console.log("3. Logaritmo em qualquer base");
        console.log("0. Voltar");
        
        opcao = await perguntar("Escolha: ");
        
        if (opcao === '1') {
            let numero = parseFloat(await perguntar("Digite o número: "));
            if (numero <= 0) {
                console.log("Erro: Logaritmo de número não positivo!");
            } else {
                console.log(`ln(${numero}) = ${Math.log(numero).toFixed(4)}`);
            }
        } else if (opcao === '2') {
            let numero = parseFloat(await perguntar("Digite o número: "));
            if (numero <= 0) {
                console.log("Erro: Logaritmo de número não positivo!");
            } else {
                console.log(`log10(${numero}) = ${Math.log10(numero).toFixed(4)}`);
            }
        } else if (opcao === '3') {
            let numero = parseFloat(await perguntar("Digite o número: "));
            let base = parseFloat(await perguntar("Digite a base: "));
            if (numero <= 0 || base <= 0 || base === 1) {
                console.log("Erro: Parâmetros inválidos!");
            } else {
                console.log(`log${base}(${numero}) = ${(Math.log(numero) / Math.log(base)).toFixed(4)}`);
            }
        }
        
        if (opcao !== '0') {
            await perguntar("\nPressione Enter para continuar...");
        }
    } while (opcao !== '0');
}

async function menuComplexos() {
    let opcao;
    do {
        console.clear();
        console.log("--- NÚMEROS COMPLEXOS ---");
        console.log("1. Adição");
        console.log("2. Multiplicação");
        console.log("3. Módulo");
        console.log("4. Forma Polar");
        console.log("0. Voltar");
        
        opcao = await perguntar("Escolha: ");
        
        if (opcao === '1') {
            let real1 = parseFloat(await perguntar("Parte real do primeiro: "));
            let imag1 = parseFloat(await perguntar("Parte imaginária do primeiro: "));
            let real2 = parseFloat(await perguntar("Parte real do segundo: "));
            let imag2 = parseFloat(await perguntar("Parte imaginária do segundo: "));
            console.log(`Resultado: (${real1} + ${imag1}i) + (${real2} + ${imag2}i) = ${real1 + real2} + ${imag1 + imag2}i`);
        } else if (opcao === '2') {
            let real1 = parseFloat(await perguntar("Parte real do primeiro: "));
            let imag1 = parseFloat(await perguntar("Parte imaginária do primeiro: "));
            let real2 = parseFloat(await perguntar("Parte real do segundo: "));
            let imag2 = parseFloat(await perguntar("Parte imaginária do segundo: "));
            let real = real1 * real2 - imag1 * imag2;
            let imag = real1 * imag2 + imag1 * real2;
            console.log(`Resultado: (${real1} + ${imag1}i) × (${real2} + ${imag2}i) = ${real} + ${imag}i`);
        } else if (opcao === '3') {
            let real = parseFloat(await perguntar("Parte real: "));
            let imag = parseFloat(await perguntar("Parte imaginária: "));
            let modulo = Math.sqrt(real * real + imag * imag);
            console.log(`Módulo de ${real} + ${imag}i = ${modulo.toFixed(4)}`);
        } else if (opcao === '4') {
            let real = parseFloat(await perguntar("Parte real: "));
            let imag = parseFloat(await perguntar("Parte imaginária: "));
            let modulo = Math.sqrt(real * real + imag * imag);
            let fase = Math.atan2(imag, real) * 180 / Math.PI;
            console.log(`Forma polar: ${modulo.toFixed(4)} ∠ ${fase.toFixed(2)}°`);
        }
        
        if (opcao !== '0') {
            await perguntar("\nPressione Enter para continuar...");
        }
    } while (opcao !== '0');
}

async function main() {
    let opcao;
    do {
        console.clear();
        mostrarMenuPrincipal();
        opcao = await perguntar("Escolha uma opção: ");
        
        if (opcao === '1') {
            await menuBasicas();
        } else if (opcao === '2') {
            await menuPotenciaRaiz();
        } else if (opcao === '3') {
            await menuTrigonometria();
        } else if (opcao === '4') {
            await menuLogaritmos();
        } else if (opcao === '5') {
            await menuComplexos();
        } else if (opcao === '0') {
            console.log("\nSaindo... Obrigado por usar a calculadora!");
        } else {
            console.log("Opção inválida!");
            await perguntar("Pressione Enter para continuar...");
        }
    } while (opcao !== '0');
    
    rl.close();
}

main();